app/Http/Controllers/AdminDash.php

```diff
@@ -1379,6 +1386,24 @@ public function saveNewEvent(Request $request) {
                 $time.'.'.$ext
             );
             $public_url = '/storage/event_banners/'.$time.'.'.$ext;
+        } elseif ($request->filled('banner_url')) {
+            try {
+                $response = Http::get($request->banner_url);
+                if (!$response->successful()) {
+                    throw new ConnectionException;
+                }
+            } catch (RequestException | ConnectionException) {
+                return redirect()->back()->withErrors('The provided URL is not valid or unreachable')->withInput();
+            }
+            $imageContent = file_get_contents($request->banner_url);
+            $imageSize = getimagesizefromstring($imageContent);
+            if ($imageSize == false) {
+                return redirect()->back()->withErrors('The provided URL does not point to an image.')->withInput();
+            }
+            $time = Carbon::now()->timestamp;
+            $ext = pathinfo($request->banner_url, PATHINFO_EXTENSION);
+            file_put_contents(storage_path('app/public/event_banners/'.$time.'.'.$ext), $imageContent);
+            $public_url = '/storage/event_banners/'.$time.'.'.$ext;
         } else {
```

Comment:
One thing here that we could change that's actually better practice - if we set `$public_url = null` and initialize that before the if statement, we know it'll get set. Some compilers (Java in particular) may not let this compile if the variables are initialized only on the if-else statement and used later outside of that scope.